calico_kubernetes/calico_kubernetes.py

```diff
@@ -504,44 +520,27 @@ def _get_pod_ports(self, pod):
         return ports
 
     def _get_pod_config(self):
-        """Get the list of pods from the Kube API server."""
-        pods = self._get_api_path('pods')
-        logger.debug('Got pods %s', pods)
-
-        for pod in pods:
-            logger.debug('Processing pod %s', pod)
-            ns = pod['metadata']['namespace'].replace('/', '_')
-            name = pod['metadata']['name'].replace('/', '_')
-            if ns == self.namespace and name == self.pod_name:
-                this_pod = pod
-                break
-        else:
-            raise KeyError('Pod not found: ' + self.pod_name)
-        logger.debug('Got pod data %s', this_pod)
-        return this_pod
-
-    def _get_api_path(self, path):
-        """Get a resource from the API specified API path.
+        """Get the pod resource from the API.
+        API Path depends on the KUBE_API_ROOT, namespace, and pod_name
 
-        e.g.
-        _get_api_path('pods')
-
-        :param path: The relative path to an API endpoint.
-        :return: A list of JSON API objects
-        :rtype list
+        :return: JSON object containing the pod spec
         """
-        logger.info('Getting API Resource: %s from KUBE_API_ROOT: %s',
-                    path, KUBE_API_ROOT)
+        logger.debug('Getting API Resource: from KUBE_API_ROOT: %s',
+                     KUBE_API_ROOT)
         session = requests.Session()
-        if self.auth_token:
+        if self._api_root_secure() and self.auth_token:
             session.headers.update({'Authorization':
                                     'Bearer ' + self.auth_token})
-        response = session.get(KUBE_API_ROOT + path, verify=False)
-        response_body = response.text
+        try:
+            path = '%snamespaces/%s/pods/%s' % (KUBE_API_ROOT, self.namespace, self.pod_name)
```

Comment:
Additionally, I think we need to be careful about how we're generating the path.

The way this is currently done, KUBE_API_ROOT _requires_ a trailing `/` - this could cause unexpected problems.  We should probably be resilient to both `http://api.url/api/v1` as well as `http://api.url/api/v1/`